### DISCUSSION TITLE
Does @commercetools-uikit/text-input support keydown events?

### DISCUSSION THREAD

**Discussion by myrdstom** [commercetools/ui-kit#1808] [Q&A] [answered]
Abit new to the commerce tools and haven't seen and keypress/keydown events included in the text-input props. Just wondering whether I can implement such a functionality if for example I want to search on `enter` press

**Comment by adnasa** [comment, marked as answer]
Hello @myrdstom 
thanks for your question.

Adding support for `onKeyDown` on `text-input` would be something I welcome a PR on.
However the handler for `enter` keyCode should be handled on the client-side.

you can go ahead and add (optional) `onKeyDown` if you want here.
https://github.com/commercetools/ui-kit/blob/master/packages/components/inputs/text-input/src/text-input.js#L43

where we simply forward the handler from the client with no proxy functionality

```jsx
  onKeyDown={props.onKeyDown}
```
